graylog2-server/src/main/java/org/graylog2/security/JwtBearerTokenProvider.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.security;
+
+import com.google.common.base.Suppliers;
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+
+import javax.crypto.spec.SecretKeySpec;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import java.nio.charset.StandardCharsets;
+import java.security.Key;
+import com.github.joschi.jadconfig.util.Duration;
+import java.util.Date;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+@Singleton
+public class JwtBearerTokenProvider implements Provider<String> {
```

Comment:
I think we can make the naming more precise here. It is not really relevant that the JWT is used as a bearer token, but I think it is relevant that it is used for Data Node auth. So maybe something like:
```suggestion
public class DataNodeAuthTokenProvider implements Provider<String> {
```

Reply:
renamed to 'indexer` as of our discussion